### PR TITLE
Add qtawesome recipe

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
this is a pyqt4 failure on linux only. cc @ccordoba12

**Comment by SylvainCorlay** [comment]
- I skipped the import on linux. 
 - The failure is that `PyQt4.QtGui` cannot be imported on linux, although I can't reproduce the issue locally.
 - I opened https://github.com/conda-forge/qtpy-feedstock/pull/1 which exposes the issue on upstream.

**Comment by jakirkham** [comment]
Maybe try copying and concatenating any `yum_requirements.txt` files from dependencies.

**Comment by SylvainCorlay** [comment]
It does not seem that there is any. Besides, there is no public recipe for pyqt4.

**Comment by jakirkham** [comment]
There is. It's called [`pyqt`]( https://github.com/conda-forge/pyqt-feedstock ) though.

**Comment by SylvainCorlay** [comment]
I opened a PR to check the import of `QtGui` in the PyQt4 recipe. I think that as far as qtawesome is concerned, it is an upstream issue though, as per conda-forge/qtpy-feedstock#1 .

**Comment by jakirkham** [comment]
Is the issue present in `pyqt` as well?

**Comment by SylvainCorlay** [comment]
It seems so, although I can't reproduce it locally.

**Comment by SylvainCorlay** [comment]
I confirm that the linux pyqt package from conda-forge is the culprit. I can reproduce the issue locally. (Just try importing `PyQt4.QtGui.QSessionManager`.)

**Comment by ccordoba12** [comment]
The error seems real. Unfortunately, I'm traveling right now, so I don't have time to look at it in detail.

I think the cause is that some PyQt objects don't get build in CircleCI. I say it because the problem is not that QtGui can't be imported per se, but that there's no `QtGui.QSessionManager` in the conda-forge package, which (to me) is related to the lack of PAM and/or console-kit in docker image.

**Comment by SylvainCorlay** [comment]
@jakirkham I think that since it is broken, we should remove the `pyqt` package from the linux conda-forge channel.

**Comment by ccordoba12** [comment]
> I think the cause is that some PyQt objects don't get build in CircleCI

Actually, that's wrong. The problem must be in Qt because PyQt only create bindings for what's present (or not) in Qt. So I have to go through the Qt logs in CircleCI and see if it asks something related to PAM and add that to `yum_requirements`.

----

> I think that since it is broken, we should remove the pyqt package

@SylvainCorlay, I think you don't understand what's at stake here. Conda-forge has a different set of dependencies on Qt than defaults, so we need to preserve both Qt and PyQt in conda-forge.

----

@jakirkham, please don't merge this until I have a chance to see what's happening with Qt. In any case, this is not urgent because `qtawesome` is in defaults and it's well maintained there because it's a Spyder dependency :-)

**Comment by SylvainCorlay** [comment]
> @SylvainCorlay, I think you don't understand what's at stake here. Conda-forge has a different set of dependencies on Qt than defaults, so we need to preserve both Qt and PyQt in conda-forge.

No problem. would it work to remove both qt and pyqt from the linux conda-forge channel until this is fixed?

**Comment by ccordoba12** [comment]
> would it work to remove both qt and pyqt from the linux conda-forge channel until this is fixed?

Nop. There are already other packages built on top of them (as @ceholden can testify :-).

I promise I'll fix this next week, I'm just taking my first vacations in two years now :-)

**Comment by jakirkham** [comment]
> please don't merge this until I have a chance to see what's happening with Qt.

Have you had a chance to understand what is going on with Qt? I saw a lot of activity and unfortunately don't remember if it was related to this or something else. Sorry.

**Comment by ccordoba12** [comment]
I just merged conda-forge/pyqt-feedstock#7 that should fix the problem reported by @SylvainCorlay here.

So as soon as pyqt build **1** is present in conda-forge, you just have to restart this PR again before merging :-)

**Comment by SylvainCorlay** [comment]
Done! (and rebased on master). It seems to work now.

Thanks @ccordoba12 .

**Comment by ccordoba12** [comment]
@ocefpaf, could you merge this one? It's ready and useful for Spyder 3.0 :-)

File: recipes/qtawesome/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+test:
+    requires:
+        - pyqt
+    imports:
+        - qtawesome
+
+about:
+    home: https://github.com/spyder-ide/qtawesome
+    license: MIT
```
Comment (jakirkham) [inline recipes/qtawesome/meta.yaml]: Please make sure to add the license file once available.

xref: https://github.com/spyder-ide/qtawesome/issues/51

**Pull request merged**